pennylane_lightning/core/src/simulators/lightning_qubit/measurements/MeasurementsLQubit.hpp

```diff
@@ -128,10 +131,14 @@ class Measurements final
             Gates::getIndicesAfterExclusion(wires, num_qubits), num_qubits);
         const std::size_t n_probs = PUtil::exp2(n_wires);
         std::vector<PrecisionT> probabilities(n_probs, 0);
+        auto *probs = probabilities.data();
         std::size_t ind_probs = 0;
         for (auto index : all_indices) {
```

Comment:
Is it not possible to include this for loop in the pragma?

Reply:
Yes, but it is typically somewhat small. Might be good to put on the inside too. Few cases to check here.

Reply:
Doing a quick test, switching the loops to make them not "perfectly nested". We get better parallel efficiency but worse performance.
![efficiency_vs_nthreads_bad](https://github.com/user-attachments/assets/d5997461-79d8-4154-a42e-9bf9962ed4fa)